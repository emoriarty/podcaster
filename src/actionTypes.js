export const FETCH_COUNTRIES = 'itunes/FETCH_COUNTRIES';
export const FETCH_COUNTRIES_SUCCESS = 'itunes/FETCH_COUNTRIES_SUCCESS';
export const FETCH_COUNTRIES_FAIL = 'itunes/FETCH_COUNTRIES_FAIL';

export const FETCH_MEDIA_TYPES = 'itunes/FETCH_MEDIA_TYPES';
export const FETCH_MEDIA_TYPES_SUCCESS = 'itunes/FETCH_MEDIA_TYPES_SUCCESS';
export const FETCH_MEDIA_TYPES_FAIL = 'itunes/FETCH_MEDIA_TYPES_FAIL';
