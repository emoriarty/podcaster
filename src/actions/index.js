export { default as fetchCountries } from './itunes/fetchCountries';
export { default as fetchMediaTypes } from './itunes/fetchMediaTypes';
