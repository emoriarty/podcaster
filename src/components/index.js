export * from './Expansion'
export * from './Layouts'
export * from './ListItems'
export * from './Texts'
