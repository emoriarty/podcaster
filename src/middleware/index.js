export { default as yqlMiddleware } from './yql'
export { default as rssMiddleware } from './rss'
