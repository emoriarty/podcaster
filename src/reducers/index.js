import { combineReducers } from 'redux';
import countries from './countries';

const app = combineReducers({
  countries
});

export default app;
