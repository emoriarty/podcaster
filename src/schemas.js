import { schema } from 'normalizr'

export const country = new schema.Entity(
  'countries',
  {},
  { idAttribute: 'country_code' }
)

export const mediaType = new schema.Entity('mediaTypes')

export const podcast = new schema.Entity(
  'podcasts',
  {},
  {
    idAttribute: (value, parent, key) => {
      return value.id.attributes['im:id']
    }
  })

export const podcastFeed = { feed: { entry: new schema.Array(podcast) } }
