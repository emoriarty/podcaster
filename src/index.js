import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import appReducer from './reducers';
import { fetchCountries } from './actions';

const loggerMiddleware = createLogger();

window.addEventListener('load', () => {
  // jsonp(
  //   'https://itunes.apple.com/search?term=carne&media=podcast',
  //   (err, response) => console.log(err, response)
  // );

  let store = createStore(
    appReducer,
    applyMiddleware(
      thunkMiddleware,
      loggerMiddleware
    )
  );

  console.log('initialState', store.getState());

  const unsubscribe = store.subscribe(() => console.log(store.getState()));

  store
    .dispatch(fetchCountries())
    .then(data => {
      console.log('success', data);
    })
    .catch(reason => {
      console.error(reason);
    });

  unsubscribe();
});
