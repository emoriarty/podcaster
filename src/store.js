import { createStore, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { createLogger } from 'redux-logger'
import { yqlMiddleware } from './middleware'
import appReducer from './reducers'

const loggerMiddleware = createLogger()

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

export default () =>
  createStore(
    appReducer,
    composeEnhancers(
      applyMiddleware(
        thunkMiddleware,
        yqlMiddleware,
        loggerMiddleware
      )
    )
  )
