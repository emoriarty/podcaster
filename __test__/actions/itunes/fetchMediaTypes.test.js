import { fetchMediaTypes } from '../../../src/actions';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([ thunk ]);

describe('Fetch media types from itunes', () => {
  it('creates FETCH_MEDIA_TYPES_SUCCESS when fetching media types', () => {
    const store = mockStore({ mediaTypes: [] });

    return store.dispatch(fetchMediaTypes())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
