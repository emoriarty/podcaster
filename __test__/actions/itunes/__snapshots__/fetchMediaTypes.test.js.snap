// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch media types from itunes creates FETCH_MEDIA_TYPES_SUCCESS when fetching media types 1`] = `
Array [
  Object {
    "type": "itunes/FETCH_MEDIA_TYPES",
  },
  Object {
    "payload": Array [
      Object {
        "canBeExplicit": true,
        "feed_types": Array [
          Object {
            "translation_key": "topaudiobooks",
            "urlPrefix": "https://itunes.apple.com/<%= country_code %>/rss/topaudiobooks/<%= parameters %>",
            "urlSuffix": "xml",
          },
        ],
        "id": "50000024",
        "store": "music",
        "subgenres": Array [
          Object {
            "id": "",
            "translation_key": "all",
          },
          Object {
            "id": "50000041",
            "translation_key": "arts&entertainment",
          },
        ],
        "translation_key": "audiobooks",
      },
    ],
    "type": "itunes/FETCH_MEDIA_TYPES_SUCCESS",
  },
]
`;
